### DISCUSSION TITLE
Filemaker Pro Integration

### DISCUSSION THREAD

**Discussion by raftopouloss** [deepgram/community#460] [General help]
Has anyone managed to integrate Deepgram with filemaker pro. It has the option of using CURL but my experience with this is limited. I have managed to integrate with Whisper using their CURL API request but not with deepgram. Any assistance would be very much appreciated.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by raftopouloss** [reply]
The curl options is:
" --request " & Quote("POST") & 
" --header " & Quote ( "accept: application/json" ) &
" --header " & Quote ( "content-type: application/json") &
" --header " & Quote ( "Authorization: Token xxxxxxxxxxxxxxxxxxxxxxx") &
" --data " & Quote ( "
{
  url: " & $audioFileName & "
}")

and the response is:
{"err_code":"Bad Request","err_msg":"Content-type was application/json, but we could not process the JSON payload.","request_id":"4aad892d-02c1-40ff-81b8-f8b4e6a0af6c"}

**Reply by raftopouloss** [reply]
Okay so I have it working when I point to the sample file using 
"https://static.deepgram.com/examples/interview_speech-analytics.wav"

But how do I get it to point to a local file:

"    --request POST ¶" &
"     --header " & Quote ( "Authorization: Token XXXXXXXXXXXXXXXXXX" ) & "¶" &
"     --header " & Quote ( "accept: application/json" ) & "¶" &
"     --header " & Quote ( "content-type: application/json" ) & "¶" &

"     --data " & Quote ("{
  \"url\": \"Macintosh HD/Users/raft/Downloads/92521.m4a\"
}")

**Comment by jkroll-deepgram** [comment]
Hi @raftopouloss, did you get this working? One detail to note is that when you want to provide a local file, you use the `--data-binary` parameter rather than `--data` with a URL, and prefix the filename with an `@` symbol. I'm not sure the exact syntax here with Filemaker Pro, but along the lines of:
```
" --data-binary " & Quote ( "@Macintosh HD/Users/raft/Downloads/92521.m4a" )
```

For reference, assuming you've downloaded the test file from https://static.deepgram.com/examples/interview_speech-analytics.wav, the plain curl command for transcribing a local file would be:
```
curl \
-X POST \
-H "Authorization: Token <your API key>" \
-H "content-type: audio/wav" \
--data-binary @interview_speech-analytics.wav \
"https://api.deepgram.com/v1/listen"
```
